### PR TITLE
update tendermint config v2

### LINKED ISSUE #1577: remove `tendermint_mode` global arg
this is redundant arg and can be set via config or env var that overrides config

### PR DESCRIPTION
**Pull request by Fraccaman**
Based on #1476 and #1483. 

closes #1577

### CODE REVIEW

File: apps/src/bin/namada-node/cli.rs [hunk 1]
```diff
@@ -7,15 +7,19 @@ use namada_apps::node::ledger;
 
 pub fn main() -> Result<()> {
     let (cmd, mut ctx) = cli::namada_node_cli()?;
+    // fixme: not sure why this is set here, since it's settable in the config
+    // file
```
Comment (tzemanovic) [inline apps/src/bin/namada-node/cli.rs:11]: agree this is redundant. Let's remove it from global args

Comment (tzemanovic) [reply apps/src/bin/namada-node/cli.rs:11]: https://github.com/anoma/namada/issues/1577

File: apps/src/bin/namada-node/cli.rs [hunk 2]
```diff
@@ -7,15 +7,19 @@ use namada_apps::node::ledger;
...
     }
     match cmd {
         cmds::NamadaNode::Ledger(sub) => match sub {
             cmds::Ledger::Run(cmds::LedgerRun(args)) => {
                 let wasm_dir = ctx.wasm_dir();
                 sleep_until(args.start_time);
-                ctx.config.ledger.tendermint.tx_index = args.tx_index;
+                // fixme: not sure why this is set here, since it's settable in
+                // the config file ctx.config.ledger.tendermint.
+                // tx_index = args.tx_index;
```
Comment (tzemanovic) [inline apps/src/bin/namada-node/cli.rs:22]: we should remove the CLI arg since it's not being used

File: apps/src/lib/client/utils.rs [hunk 3]
```diff
@@ -324,26 +324,23 @@ pub async fn join_network(
...
-                    {
-                        node_id != *peer_id
-                    } else {
-                        true
-                    }
-                });
+            // fixme: we shouldn't be setting config values that are settable in
+            // the actual config.toml config.ledger.
+            // tendermint_config.p2p.pex = false; Remove self from
+            // persistent peers
```
Comment (tzemanovic) [inline apps/src/lib/client/utils.rs:332]: I think pex should still be switched off for validators by default - any reason to comment it out here?

File: apps/src/lib/client/utils.rs [hunk 4]
```diff
@@ -755,57 +752,70 @@ pub fn init_network(
                             None
                         })
                     .collect();
-            config.ledger.tendermint.consensus_timeout_commit =
+            // fixme: We shouldn't be hardcoding these. The timeouts should only
+            // be set in the config files.
```
Comment (tzemanovic) [inline apps/src/lib/client/utils.rs:756]: The values are not hard-coded and they are set in the config files, no?

File: apps/src/lib/client/utils.rs [hunk 5]
```diff
@@ -755,57 +752,70 @@ pub fn init_network(
...
-                config
-                    .ledger
-                    .tendermint
-                    .p2p_address
-                    .set_ip(IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)));
+                config.ledger.cometbft.p2p.laddr = TendermintAddress::from_str(
+                    &format!("0.0.0.0:{}", first_port),
+                )
+                .unwrap();
+                // .set_ip(IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)));
```
Comment (tzemanovic) [inline apps/src/lib/client/utils.rs:774]:
```suggestion
```

File: apps/src/lib/client/utils.rs [hunk 6]
```diff
@@ -755,57 +752,70 @@ pub fn init_network(
...
-                config
-                    .ledger
-                    .tendermint
-                    .rpc_address
-                    .set_ip(IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)));
+                config.ledger.cometbft.rpc.laddr = TendermintAddress::from_str(
+                    &format!("0.0.0.0:{}", first_port + 1),
+                )
+                .unwrap();
+                // .set_ip(IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)));
```
Comment (tzemanovic) [inline apps/src/lib/client/utils.rs:784]:
```suggestion
```

File: apps/src/lib/client/utils.rs [hunk 7]
```diff
@@ -755,57 +752,70 @@ pub fn init_network(
...
-            config.ledger.shell.ledger_address.set_port(first_port + 2);
+            config.ledger.cometbft.rpc.laddr = TendermintAddress::from_str(
+                &format!("{}:{}", ip, first_port + 1),
+            )
+            .unwrap();
+            // config
+            //     .ledger
+            //     .tendermint
+            //     .rpc_address
+            //     .set_port(first_port + 1);
```
Comment (tzemanovic) [inline apps/src/lib/client/utils.rs:794]:
```suggestion
```

File: apps/src/lib/client/utils.rs [hunk 8]
```diff
@@ -755,57 +752,70 @@ pub fn init_network(
...
     // Open P2P address
     if !localhost {
-        config
-            .ledger
-            .tendermint
-            .p2p_address
-            .set_ip(IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)));
+        config.ledger.cometbft.p2p.laddr =
+            TendermintAddress::from_str("0.0.0.0:26656").unwrap();
+        // .set_ip(IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)));
```
Comment (tzemanovic) [inline apps/src/lib/client/utils.rs:816]:
```suggestion
```

File: apps/src/lib/config/mod.rs [hunk 9]
```diff
@@ -158,56 +129,32 @@ impl Ledger {
...
             shell: Shell {
                 base_dir: base_dir.as_ref().to_owned(),
-                ledger_address: SocketAddr::new(
-                    IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)),
-                    26658,
-                ),
+                // ledger_address: SocketAddr::new(
+                //     IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)),
+                //     26658,
+                // ),
```
Comment (tzemanovic) [inline apps/src/lib/config/mod.rs:146]:
```suggestion
```

File: apps/src/lib/config/mod.rs [hunk 10]
```diff
@@ -399,6 +346,18 @@ pub fn get_default_namada_folder() -> PathBuf {
     }
 }
 
+mod tests {
```
Comment (tzemanovic) [inline apps/src/lib/config/mod.rs:349]:
```suggestion
#[cfg(test)]
mod tests {
```

Comment (tzemanovic) [reply apps/src/lib/config/mod.rs:349]: also, let's move the tests to the bottom of the file

File: apps/src/lib/config/mod.rs [hunk 11]
```diff
@@ -399,6 +346,18 @@ pub fn get_default_namada_folder() -> PathBuf {
...
+
+    #![allow(unused_imports)]
```
Comment (tzemanovic) [inline apps/src/lib/config/mod.rs:351]:
```suggestion
```

File: apps/src/lib/node/ledger/tendermint_node.rs [hunk 12]
```diff
@@ -348,23 +347,24 @@ pub fn write_validator_state(home_dir: impl AsRef<Path>) {
...
-    tendermint_config: config::Tendermint,
+    config: TendermintConfig,
 ) -> Result<()> {
     let home_dir = home_dir.as_ref();
     let path = home_dir.join("config").join("config.toml");
-    let mut config =
-        TendermintConfig::load_toml_file(&path).map_err(Error::LoadConfig)?;
+    let mut config = config.clone();
+    // let mut config =
+    //     TendermintConfig::load_toml_file(&path).map_err(Error::LoadConfig)?;
```
Comment (tzemanovic) [inline apps/src/lib/node/ledger/tendermint_node.rs:356]:
```suggestion
```

File: apps/src/lib/node/ledger/tendermint_node.rs [hunk 13]
```diff
@@ -348,23 +347,24 @@ pub fn write_validator_state(home_dir: impl AsRef<Path>) {
...
-            .unwrap();
-    config.p2p.persistent_peers = tendermint_config.p2p_persistent_peers;
-    config.p2p.pex = tendermint_config.p2p_pex;
-    config.p2p.allow_duplicate_ip = tendermint_config.p2p_allow_duplicate_ip;
+    // config.p2p.laddr =
+    //     TendermintAddress::from_str(&tendermint_config.p2p_address.
+    // to_string())         .unwrap();
+    // config.p2p.persistent_peers = tendermint_config.p2p_persistent_peers;
+    // config.p2p.pex = tendermint_config.p2p_pex;
+    // config.p2p.allow_duplicate_ip = tendermint_config.p2p_allow_duplicate_ip;
```
Comment (tzemanovic) [inline apps/src/lib/node/ledger/tendermint_node.rs:368]:
```suggestion
```

File: apps/src/lib/node/ledger/tendermint_node.rs [hunk 14]
```diff
@@ -375,47 +375,47 @@ async fn update_tendermint_config(
     // again in the future.
     config.mempool.keep_invalid_txs_in_cache = false;
 
-    config.rpc.laddr =
-        TendermintAddress::from_str(&tendermint_config.rpc_address.to_string())
-            .unwrap();
+    // config.rpc.laddr =
+    //     TendermintAddress::from_str(&tendermint_config.rpc_address.
+    // to_string())         .unwrap();
```
Comment (tzemanovic) [inline apps/src/lib/node/ledger/tendermint_node.rs:380]:
```suggestion
```

File: apps/src/lib/node/ledger/tendermint_node.rs [hunk 15]
```diff
@@ -375,47 +375,47 @@ async fn update_tendermint_config(
...
+    // }
+
+    // let indexer = if tendermint_config.tx_index {
+    //     TxIndexer::Kv
+    // } else {
+    //     TxIndexer::Null
+    // };
+    // #[cfg(feature = "abcipp")]
+    // let indexer = [indexer];
+    // config.tx_index = TxIndexConfig { indexer };
```
Comment (tzemanovic) [inline apps/src/lib/node/ledger/tendermint_node.rs:419]:
```suggestion
```

File: tests/src/e2e/ledger_tests.rs [hunk 16]
```diff
@@ -3959,13 +3961,32 @@ fn test_genesis_validators() -> Result<()> {
...
-            .tendermint
-            .rpc_address
-            .set_port(first_port + 1);
-        config.ledger.shell.ledger_address.set_port(first_port + 2);
+        let p2p_addr =
+            convert_tm_addr_to_socket_addr(&config.ledger.cometbft.p2p.laddr)
+                .ip()
+                .to_string();
+        println!("ERROR: {}", p2p_addr);
+        println!("ERROR: {}", first_port);
```
Comment (tzemanovic) [inline tests/src/e2e/ledger_tests.rs:3969]:
```suggestion
```

File: tests/src/e2e/ledger_tests.rs [hunk 17]
```diff
@@ -4148,13 +4169,32 @@ fn double_signing_gets_slashed() -> Result<()> {
...
```
Comment (tzemanovic) [inline tests/src/e2e/ledger_tests.rs:4177]:
```suggestion
```

File: apps/src/lib/client/utils.rs (hunk 3 above)
Comment (Fraccaman) [reply apps/src/lib/client/utils.rs:332]: cause if we force it here, we can't modify it via config

**Comment by tzemanovic** [comment]
pls update wasm

**Review by tzemanovic — APPROVED** [review]

**Review by tzemanovic — DISMISSED** [review]

**Pull request closed without merging**